File: src/main/java/com/lewandowski/springproject/service/CompanyService.java
package com.lewandowski.springproject.service;

import com.lewandowski.springproject.domain.Company;

import java.util.List;

public interface CompanyService {

    List<Company> getAllCompanies();

    Company addCompany (Company company);

    void deleteCompany(String companyName);

    Company updateCompany(String companyName, Company company);

    Company getCompanyByName(String companyName);

}

File: projekt-frontend/src/services/CompanyService.js
import axios from 'axios';

const COMPANY_API_BASE_URL = "http://localhost:8080/api";

class CompanyService{
    getCompanies(){
        return axios.get(COMPANY_API_BASE_URL + "/company");
    }

    createCompany(company){
        return axios.post(COMPANY_API_BASE_URL + "/company", company);
    }

    updateCompany(company, companyName){
        return axios.put(COMPANY_API_BASE_URL + '/company/' + companyName, company);
    }

    deleteCompany(companyName){
        return axios.delete(COMPANY_API_BASE_URL + '/company/' + companyName);
    }

    getCompanyByName(companyName){
        return axios.get(COMPANY_API_BASE_URL + '/company/' + companyName);
    }

}
export default new CompanyService();